worker/uniter/runner/context/context.go

```diff
@@ -1147,8 +1147,10 @@ func (ctx *HookContext) doFlush(process string) error {
 // we'll still trigger a change to a counter on the podspec so that we can
 // ensure any other charm changes (eg storage) are acted on.
 func (ctx *HookContext) addCommitHookChangesForCAAS(builder *uniter.CommitHookParamsBuilder, process string) error {
-	if ctx.podSpecYaml == nil && ctx.k8sRawSpecYaml == nil && process != string(hooks.UpgradeCharm) {
-		// No ops for any situation unless any k8s spec needs to be set or "upgrade-charm" was run.
+	if ctx.podSpecYaml == nil && ctx.k8sRawSpecYaml == nil {
```

Comment:
As discussed need to fix this check so that leaders still go through will nill spec